src/devtools/client/debugger/src/components/SourceOutline/SourceOutline.tsx

```diff
@@ -47,7 +47,8 @@ export function SourceOutline({
 
   useEffect(() => {
     if (selectedSource) {
-      dispatch(setBreakpointHitCounts(selectedSource.id));
+      // We start by loading the first N lines of hits, where N is the line limit.
+      dispatch(setBreakpointHitCounts(selectedSource.id, 1));
```

Comment:
Does this mean that the hits in the Outline view will only reflect hits in lines 1-500?

Reply:
And it reads like it will update to be a more accurate number, but only if the user hovers on other lines in the editor (which forces the load of hits in other lines). Did I understand that correctly?

Reply:
So, we just talked about this at backend standup, and it brings up a great point. Here's a further improvement:

- As soon as a file loads, grab the locations that are outside of function definitions (and so ran as soon as the file loaded)
- Also grab the first location (what we use for outline view) inside of all functions and ask for the hits at each of those locations
- If any of those hits come back as 0, we don't need to fetch any further hitCounts for that function, because no later location can be non-0 if the first location is 0
- Theoretically, you could do this for each level of scope/function nesting, but that's probably a bridge farther than we need to go to for this version.

I think I have a good follow-up here, which would be to differentiate between 0 and null for the outline view, which might be less misleading. I'll do that first, then follow-up with the above points.

Reply:
OK, so I've made it so that if we aren't loading the line that that function definition falls on it will *not* have a hitCount marker. As soon as we load it we will show the hitCounts (but you have to go to that line and hover there). I am happy with this compromise because:

A) I think most of the time people will not be using these features on 1000+ line files
B) If they *are* doing that, there is still a way for them to get the info, it's just not perfect
C) All of this is behind a feature flag 😄